src/Models/ModelRelationshipDefinition.php

```diff
@@ -79,6 +97,66 @@ public function belongsTo(): self {
 		return $this;
 	}
 
+	/**
+	 * Set the callable to hydrate the relationship with.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @param callable $hydrateWith The callable to hydrate the relationship with.
+	 */
+	public function setHydrateWith( callable $hydrateWith ): self {
+		$this->checkLock();
+
+		$this->hydrateWith = $hydrateWith;
+
+		return $this;
+	}
+
+	/**
+	 * Get the callable to hydrate the relationship with.
+	 *
+	 * @since 2.0.0
+	 *
+	 * @return callable( mixed $value ): ( Model )
+	 */
+	public function getHydrateWith(): callable {
+		// By default, it returns whats given.
+		return $this->hydrateWith ?? fn( $value ) => $value;
```

Comment:
```suggestion
		return $this->hydrateWith ?? static fn( $value ) => $value;
```
Small performance improvement.